internal/agent/device/spec/spec_test.go

```diff
@@ -215,11 +217,354 @@ func TestEnsure(t *testing.T) {
 	})
 }
 
+func TestRead(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockReadWriter := fileio.NewMockReadWriter(ctrl)
+
+	s := &SpecManager{
+		log:              log.NewPrefixLogger("test"),
+		deviceReadWriter: mockReadWriter,
+	}
+
+	t.Run("bubbles up errors from readRenderedSpecFromFile", func(t *testing.T) {
+		mockReadWriter.EXPECT().ReadFile(gomock.Any()).Return(nil, errors.New("read gone wrong"))
+		_, err := s.Read(Current)
+		require.ErrorIs(err, ErrReadingRenderedSpec)
+	})
+
+	t.Run("reads a device spec", func(t *testing.T) {
+		image := "flightctl-device:v1"
+		spec, err := createTestSpec(image)
+		require.NoError(err)
+		mockReadWriter.EXPECT().ReadFile(gomock.Any()).Return(spec, nil)
+
+		specFromRead, err := s.Read(Current)
+		require.NoError(err)
+		require.Equal(image, specFromRead.Os.Image)
+	})
+}
+
+func Test_readRenderedSpecFromFile(t *testing.T) {
+	require := require.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockReader := fileio.NewMockReader(ctrl)
+	filePath := "test/path/spec.json"
+
+	t.Run("error when the file does not exist", func(t *testing.T) {
+		mockReader.EXPECT().ReadFile(filePath).Return(nil, os.ErrNotExist)
+
+		_, err := readRenderedSpecFromFile(mockReader, filePath)
+		require.ErrorIs(err, ErrMissingRenderedSpec)
+	})
+
+	t.Run("error reading file returns an error containing the file path", func(t *testing.T) {
```

Comment:
not sure this test name is accurate also we want to keep these as concise as possible.

Reply:
ahh this was leftoever from earlier testing without the shared error types, will update